src/NetscriptFunctions.ts

```diff
@@ -752,9 +752,25 @@ export function NetscriptFunctions(workerScript: WorkerScript): NS {
 
       return numCycleForGrowth(server, Number(growth), Player, cores);
     },
-    growthAnalyzeSecurity: function (_threads: unknown): number {
+    growthAnalyzeSecurity: function (_threads: unknown, _hostname: unknown, _cores?: unknown): number {
       updateDynamicRam("growthAnalyzeSecurity", getRamCost(Player, "growthAnalyzeSecurity"));
-      const threads = helper.number("growthAnalyzeSecurity", "threads", _threads);
+      let threads = helper.number("growthAnalyzeSecurity", "threads", _threads);
+      if (_hostname) {
+        const cores = helper.number("growthAnalyzeSecurity", "cores", _cores) || 1;
+        const hostname = helper.string("growthAnalyzeSecurity", "hostname", _hostname);
+        const server = safeGetServer(hostname, "growthAnalyzeSecurity");
+
+        if (!(server instanceof Server)) {
+          workerScript.log("growthAnalyzeSecurity", () => "Cannot be executed on this server.");
+          return 0;
+        }
+
+        const growthFactor = server.moneyMax / server.moneyAvailable;
+        const maxThreadsNeeded = Math.ceil(numCycleForGrowth(server, growthFactor, Player, cores));
```

Comment:
```suggestion
        const maxThreadsNeeded = Math.ceil(numCycleForGrowthCorrected(server, server.moneyMax, server.moneyAvailable, Player, cores));
```

As I see from
https://github.com/danielyxie/bitburner/blob/e1a56f9270df83a834c8e37c0de8cc7f939021c9/src/Server/ServerHelpers.ts#L303-L308
`numCycleForGrowthCorrected` is used